lib/ds/hashmap.flow

```diff
@@ -221,6 +223,18 @@ pairs2hashMapWithLoadFactor(pairs : [Pair<?, ??>], hash : (?) -> int, loadFactor
 	hashMap
 }
 
+pairs2hashMapArray(pairs : [Pair<?, [??]>], hash : (?) -> int) -> HashMap<?, [??]> {
+	len = floor(i2d(length(pairs)) / defaultHashMapLoadFactor) * 2;
+	hashMap = makeHashMapOfVolume(hash, len, defaultHashMapLoadFactor);
+	fold(pairs, hashMap, \newHashMap, pair -> { 
+		setHashMap(
+			newHashMap,
+			pair.first,
+			eitherMap(lookupHashMap(hashMap, pair.first), \arr -> concat(arr, pair.second), pair.second)
```

Comment:
lookupHashMapDef  seems like your friend: Just return the empty array as def, and then you can concat?

Reply:
lookupHashMapDef + concat produces less garbage?

Reply:
Probably similar garbage but easier to read